code/game/objects/items/devices/mirage.dm

```diff
@@ -0,0 +1,65 @@
+/obj/item/explosive/grenade/mirage
+	name = "mirage grenade"
+	desc = "A special device that, when activated, produces a pair of holographic copies of the user."
+	icon_state = "delivery"
+	item_state = "delivery"
+	///the parent to be copied
+	var/mob/living/current_user
+	///How long the illusory fakes last
+	var/illusion_lifespan = 15 SECONDS
+
+/obj/item/explosive/grenade/mirage/activate(mob/user)
+	. = ..()
+	current_user = user
+	icon_state = "delivery_active"
+
+/obj/item/explosive/grenade/mirage/prime()
+	if(current_user)
+		var/mob/living/simple_animal/hostile/illusion/M = new(get_turf(src))
+		M.copy_appearance(current_user, illusion_lifespan)
+		var/mob/living/simple_animal/hostile/illusion/I = new(get_turf(src))
+		I.copy_appearance(current_user, illusion_lifespan)
+	QDEL_NULL(src)
```

Comment:
why `QDEL_NULL` and not just `qdel`?

Reply:
because i was lazy and thats the first thing comes up when I started typing qdel, I'll change it